aptos-move/e2e-move-tests/src/tests/code_publishing.rs

```diff
@@ -287,38 +287,6 @@ fn code_publishing_arbitray_dep_different_address(#[case] features: Vec<FeatureF
     );
 }
 
-#[rstest]
```

Comment:
This looks like the same as a PR you already landed?

Reply:
No, that was to fix some code that wasn't actually being tested.

This does two things, introduces a new means to leverage a resource account that we stumbled upon when building out the NFT "airdrop" contract. Namely, that one would have to put all the application data into the contract itself, which means we cannot seamlessly allow a frontend to just deploy the new contract for any NFT drop. They would have to manually update the contract and compile it.

This also further calls to action a need to support a better interface to our from_bcs... Ideally we can empower developers to specify which data structures can be deserialized without concern to safety. I have some ideas and it seems to be the same problem we have with transaction input validation.

Reply:
Looks like you moved this over from the other PR compared to last time I looked, therefore my question

```
[davidiw](https://github.com/davidiw) changed the title [move-e2e] cleanup some test code and enhance resource account [move-e2e] cleanup some test code [12 hours ago](https://github.com/aptos-labs/aptos-core/pull/4522#event-7452931400)
```

Reply:
if users want to deploy NFT contract for a new airdrop, then they can modify the contract with a new set of configs in init_module. I can see it improving the dev efficiency but these users don't need to modify contract. But they still need to prepare new config data. would it worth of adding these function to our core contracts?